ui-app/client/src/functions/ScrollToGrid.ts

```diff
@@ -0,0 +1,45 @@
+import {
+	FunctionSignature,
+	AbstractFunction,
+	FunctionExecutionParameters,
+	FunctionOutput,
+	EventResult,
+	Event,
+	Parameter,
+	Schema,
+} from '@fincity/kirun-js';
+import { NAMESPACE_UI_ENGINE } from '../constants';
+
+const SIGNATURE = new FunctionSignature('ScrollToGrid')
+	.setNamespace(NAMESPACE_UI_ENGINE)
+	.setParameters(
+		new Map([
+			Parameter.ofEntry('gridkey', Schema.ofString('gridkey')),
+			Parameter.ofEntry(
+				'behaviour',
+				Schema.ofString('behaviour')
+					.setEnums(['Instant', 'Smooth'])
+					.setDefaultValue('Instant'),
+			),
+		]),
+	)
+	.setEvents(new Map([Event.eventMapEntry(Event.OUTPUT, new Map())]));
+
+export class ScrollToGrid extends AbstractFunction {
+	protected async internalExecute(context: FunctionExecutionParameters): Promise<FunctionOutput> {
+		const gridkey: string = context.getArguments()?.get('gridkey');
+		const behaviour: string = context.getArguments()?.get('behaviour');
+		const gridElement = document.getElementById(gridkey);
+		if (gridElement) {
```

Comment:
Stop using chatGPT ;) you can use optional chaining operator before calling the function. You don't need if